src/resource/archive/ArchiveManager.cpp

```diff
@@ -143,10 +143,12 @@ void ArchiveManager::ResetVirtualFileSystem() {
 bool ArchiveManager::WriteFile(std::shared_ptr<Archive> archive, const std::string& filename,
                                const std::vector<uint8_t>& data) {
     if (archive) {
-        archive->WriteFile(filename, data);
-        return true;
+        if (archive->WriteFile(filename, data)) {
+            ResetVirtualFileSystem();
```

Comment:
Instead of resetting the VFS, you can probably just add the new file to the VFS.

Reply:
What if the file already exists?

Reply:
And how do I add the new file?

Reply:
You can add the file by adding it to the VFS map. If it already exists in the VFS no need to do anything.